posts/coding-intentionally-in-bash-grains.md

```diff
@@ -0,0 +1,194 @@
+
+Anytime there's more than one person working on a piece of code, the amount of care and deliberateness required to keep things sane goes *way* up.  It's no longer the concept that lives in your brain and the code that just has to make that concept happen.  Now, the concept has to live *inside the code* where all the collaborators can see it and change it if need be.
+
+*How* you go about implementing something doesn't mean much to the end user, but it should speak *volumes* to every engineer that touches your design at any point.  There are often a lot of ways to achieve the same functionality, and it might seem like any of the options would be sufficient to get the job done.  However, I believe that every single decision you make should have a reason (even if it's a small decision with a small reason), and that reason should communicate an objective or requirement.
+
+The idea that implementation details should help readers of code discern thought process, goals, and priorities is called **design intent**.  How you name your variables, what parameters your function takes, and how things are abstracted are all places where the design intent can be expressed — either well or poorly.
+
+I'm a firm believer that design intent is one of the most important things to consider when implementing an engineering design.  It's one of the things that sets Software Engineering apart from programming.  
+
+ > Software engineering is what happens to programming when you add time and other programmers.
+ >
+ > — [Russ Cox](https://research.swtch.com/vgo-eng)
+
+## Design intent is cross-discipline
+
+I work as a mechanical engineer, designing [injection molds](https://youtu.be/WHwTHarf8Ck?t=51), mostly for medical devices.  All of my designs, once they're finished, go straight out the door into the machine shop, where they start making all the pieces and assembling them together.  Since they don't know everything that went through my head as I was creating each design, I have to find a way to *show* my intent through the design itself.
+
+A lot of times, there are features that are especially critical.  Either the customer has said that they need special tight tolerances there, or the way the mold fits together requires extreme accuracy for some reason.  So, in order to help the machinists create the pieces in a way that will prioritize accuracy on the important bits, I have to leave spots that are specifically square or easy to put in a vice a particular way.  That way, the easiest path for them produces the best results for me.
+
+There are also spots where the dimensions aren't as critical.  For example, if I put a hole in the design that's just for an air vent, I'll make it a nice common size like 1/4", which is .250".  
+
+>  I'll put a parallel metric example in parenthesis (6mm).  The numbers aren't exactly the same, but the illustrational effect will be similar.
```

Comment:
Can we keep it to only metric, per https://github.com/exercism/blog/pull/5#discussion_r255150385?

Reply:
Yeah, I thought that I had seen a comment like that, but when I went back and looked, I couldn't find it so I thought maybe you changed your mind.

That's fine.  It's not my favorite, because we work in inches about 90% and dip into millimeters only when we have to, but I can see the reasoning for simplifying the article and taking the focus off of the units, and in theory, there's a larger portion of the readers that would feel more comfortable in metric.  That's what you get when you work with a bunch of grumpy, American machinists I suppose :grimacing: 

I'll make the switch to the *subpar* unit system.  :stuck_out_tongue: